src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,143 @@
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Performance,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Check that the object is a StringBuilder
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                IMethodSymbol appendStringMethod = stringBuilderType
+                    .GetMembers("Append")
+                    .OfType<IMethodSymbol>()
+                    .WhereAsArray(s =>
+                        s.Parameters.Length == 1 &&
+                        s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                    .FirstOrDefault();
+                if (appendStringMethod is null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    IInvocationOperation invocationOperation = (IInvocationOperation)operationContext.Operation;
+                    if (invocationOperation.Arguments.Length < 1)
+                    {
+                        return;
+                    }
+
+
+                    if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+                    IArgumentOperation firstArgument = arguments[0];
+
+                    // We don't handle class fields. Only local declarations
```

Comment:
I think the analyzer can be simplified quite a bit here:

``` csharp
   if (!TryGetConstChar(arguments[0], out char constChar))
   {
      return;
   }

   if (arguments[0] is ILocalReferenceOperation localReference)
   {
      operationContext.ReportDiagnostic(localReference.Local.CreateDiagnostic(Rule));
   }
   else
   {
      operationContext.ReportDiagnostic(arguments[0].CreateDiagnostic(Rule));
   }
}

// internal so it can be accessed by the code fixer
internal static bool TryGetConstChar(IOperation operation, out char constChar)
{
   if (argument.ConstantValue.HasValue && argument.ConstantValue.Value is string unitString && unitString.Length == 1)
   {
      constChar = unitString[0];
      return true;
   }

   constChar = default;
   return false;
}   
```